ui/src/component/diverging-stacked.js

```diff
@@ -0,0 +1,91 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import ReactEcharts from 'echarts-for-react';
+import _ from 'lodash';
+import { validate } from '../utils';
+
+export default class DivergingStacked extends PureComponent {
+  static propTypes = {
+    source: PropTypes.arrayOf(PropTypes.array.isRequired).isRequired,
+  }
+
+  render() {
+    // e.g. [ Date, Negative, Neutral, Positive ]
+    const { source } = this.props;
+    validate(source);
+
+    const dimensions = _.first(source);
+    if (_.size(dimensions) !== 4) {
+      throw new Error('DivergingStacked chart data source should have 4 columns, e.g. Date, Negative, Neutral, Positive');
```

Comment:
Assuming 4 columns here seems a bit limited.
Take the first result from google as an example:
![chart](https://peltiertech.com/images/2016-08/DivergeLiars4.png)

From this picture we have:
1. Y axis is not date
2. Not only 3 legends
3. Zero of x axis is shown

Reply:
Above snapshot in our PDF is called 'Spine', which has already been checked in. 

This chart is used to show the distribution of attitude, which is all positive value, and the center of neutral is one same x point.